### PR TITLE
[release/v2.26] Setup Kubevirt provider network controller in the seed-controller-manager

### PR DESCRIPTION
**Pull request by cnvergence**
**What this PR does / why we need it**:
Manual cherry-pick of https://github.com/kubermatic/kubermatic/pull/13858
This PR sets up a new controller that will reconcile clusters and configure KKP deployments with specific provider network requirements.
First use case will be setting up cluster isolation between customer networks.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Setup KubeVirt network controller in the seed-controller-manager.
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -390,3 +393,39 @@ require (
...
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.31.2
+	k8s.io/kubectl => k8s.io/kubectl v0.31.2
+	k8s.io/kubelet => k8s.io/kubelet v0.31.2
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.31.2
+	k8s.io/metrics => k8s.io/metrics v0.31.2
+	k8s.io/mount-utils => k8s.io/mount-utils v0.31.2
+	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.31.2
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.31.2
+	kubevirt.io/client-go => github.com/kubeovn/kubevirt-client-go v0.0.0-20240823060554-65405ba5499d
+)
```
Comment (embik) [inline go.mod:431]: This does not look at all. Do we really have to override a substantial amount of our dependencies?

Comment (cnvergence) [reply go.mod:431]: I suppose kube-ovn is problematic here https://github.com/kubermatic/kubermatic/pull/13858/commits/b0bd63322754f04a77be9798182ee9c10d7bff23

**Review by embik — APPROVED** [review]
/approve

I'm really not comfortable with this code, but I guess there's not much choice to be had here. Please, someone clean this up in `main` and on `release/v2.26` and use unstructured types. The only reason we seem to be importing `kubeovn` is for fetching the `Subnet` resource one time, which could be done with unstructured objects that don't pollute our `go.mod` with replace directives.

**Comment by cnvergence** [comment]
I agree with you, we could also bring this Subnet CRD in and maintain it 
But this will come up as a follow-up @embik

**Pull request merged**